.circleci/RoboFile.php

```diff
@@ -42,6 +51,165 @@ class RoboFile extends Tasks {
+      ->accessToken(getenv('GITHUB_TOKEN'))
+      ->uri('SU-HSDO/suhumsci')
+      ->description("Release $version")
+      ->changes($changes)
+      ->run();
+
+    if (!$result->wasSuccessful()) {
+      throw new Exception('Release was unsuccessful');
+    }
+  }
+
+  /**
+   * Get the last version from the profile.
+   *
+   * @return string
+   *   Last semver version.
+   */
+  protected function getLastVersion() {
+    $profile_info = Yaml::parseFile(__DIR__ . '/docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.info.yml');
+    return $profile_info['version'];
+  }
+
+  /**
+   * Update the changelog file with the given changes.
+   *
+   * @param string $version
+   *   New version.
+   * @param array $changes
+   *   Array of change strings.
+   */
+  protected function updateChangelog($version, array $changes = []) {
+    $divider = str_repeat('-', 80);
+    $this->taskChangelog(__DIR__ . '/docs/CHANGELOG.md')
+      ->setHeader("$version\n{$divider}\n_Release Date: " . date("Y-m-d") . "_\n\n")
+      ->anchor("# HumSci")
+      ->setBody(implode("\n", $changes))
+      ->setVerbosityThreshold(VerbosityThresholdInterface::VERBOSITY_VERBOSE)
+      ->run();
+  }
+
+  /**
+   * Advance to the next SemVer version.
+   *
+   * The behavior depends on the parameter $stage.
+   *   - If $stage is empty, then the patch or minor version of $version is
+   *     incremented
+   *   - If $stage matches the current stage in the current version, then add
+   *     one to the stage (e.g. alpha3 -> alpha4)
+   *   - If $stage does not match the current stage in the current version, then
+   *     reset to '1' (e.g. alpha4 -> beta1)
+   *
+   * Taken from consolidation/robo library.
+   *
+   * @param string $version
+   *   A SemVer version.
+   * @param string $stage
+   *   Release stage: dev, alpha, beta, rc or an empty string for stable.
+   *
+   * @return string
+   *   New semver version.
+   */
+  protected function incrementVersion($version, $stage = '') {
+    $stable = empty($stage);
+
+    preg_match('/-([a-zA-Z]*)([0-9]*)/', $version, $match);
+    $match += ['', '', ''];
+    $versionStage = $match[1];
+    $versionStageNumber = $match[2];
+    if ($versionStage != $stage) {
+      $versionStageNumber = 0;
+    }
+    $version = preg_replace('/-.*/', '', $version);
+    $versionParts = explode('.', $version);
+    if ($stable) {
+      $versionParts[count($versionParts) - 1]++;
+    }
+    $version = implode('.', $versionParts);
+    if (!$stable) {
+      $version .= '-' . $stage;
+      if ($stage != 'dev') {
+        $versionStageNumber++;
+        $version .= $versionStageNumber;
+      }
+    }
+    return $version;
+  }
+
+  /**
+   * Set the versions in the module & profile info files.
+   *
+   * @param string $version
+   *   New version.
+   */
+  protected function setVersions($version) {
+    // Update only modules in humsci and the appropriate profiles.
+    $dirs = [
```

Comment:
Oh interesting, you are also updating the 'custom' modules that are in this repo. 🤔 I like that it is automated, I'm not sure about the coupling of versions.

Reply:
i've been manually updating the humsci modules to be the same as the profile. :man_shrugging: